dask_ml/model_selection/_incremental.py

```diff
@@ -614,7 +646,34 @@ def fit(self, X, y=None, **fit_params):
         **fit_params
             Additional partial fit keyword arguments for the estimator.
         """
-        return default_client().sync(self._fit, X, y, **fit_params)
+        if isinstance(self.verbose, float) and not (0 < self.verbose <= 1):
+            msg = "verbose={} must satisfy 0 < verbose <= 1"
+            raise ValueError(msg.format(self.verbose))
+
+        if isinstance(self.verbose, bool):
+            # Log all the time if False or True.
+            # If True, logger will be configured to pipe to stdout
+            verbosity = 1
+        elif isinstance(self.verbose, float):
+            verbosity = np.round(1 / self.verbose)
+        else:
+            msg = "verbose must be an instance of float or bool. Got type={}"
+            raise TypeError(msg.format(type(self.verbose)))
+
+        kwargs = dict(verbosity=verbosity, **fit_params)
+        if self.verbose:
+            logger.setLevel(logging.INFO)
+            h = logging.StreamHandler(sys.stdout)
+            self._h = h
+            self._logging_context = LoggingContext(
```

Comment:
Do we need to assign this? Why can't we just do `with LoggingContext()...`?

Also, let's keep the call to `sync` the same.

```
if self.verbose:
    context = LoggingContext(...)
else:
    context = contextlib.nullcontext()

with context:
    return default_client().sync(...)
```

Reply:
That's a lot simpler!